model/loss/base_loss.py

```diff
@@ -557,3 +557,136 @@ def call(self, y_true: Dict[Text, tf.Tensor],
         weights)
     # Reduce_sum over the channels (i.e., number of masks).
     return tf.reduce_sum(weighted_dice_losses, axis=-1)
+
+
+class SILogError(tf.keras.losses.Loss):
+  """This class contains code to compute the SILog error.
+
+  Scale invariant logarithmic (SILog) error was proposed for monocular depth
+  estimation.
+
+  Reference:
+  Eigen, David, Christian Puhrsch, and Rob Fergus. "Depth map prediction from a
+  single image using a multi-scale deep network." In NeurIPS, 2014.
+  """
+
+  def __init__(self,
+               gt_key: Text,
+               pred_key: Text):
+    # Implicit reduction might mess with tf.distribute.Strategy, hence we
+    # explicitly reduce the loss.
+    super().__init__(reduction=tf.keras.losses.Reduction.NONE)
+    self._gt_key = gt_key
+    self._pred_key = pred_key
+
+  def call(self, y_true: Dict[Text, tf.Tensor],
+           y_pred: Dict[Text, tf.Tensor]) -> tf.Tensor:
+    """Computes the scale invariant logarithmic error.
+
+    Args:
+      y_true: A dict of tensors providing ground-truth information.
+      y_pred: A dict of tensors providing predictions.
+
+    Returns:
+      A tensor of shape [batch] containing the loss per sample.
+    """
+    gt = y_true[self._gt_key]
+    pred = y_pred[self._pred_key]
+
+    def _compute_error(loss_input):
+      gt, pred = loss_input
+      # Unlabeled pixels are filled with 0.
+      label_mask = gt > 0
```

Comment:
Maybe share this depth ignore 0 with all the other codes (e.g., define it in common.py)